gameplay/src/Ref.h

```diff
@@ -43,6 +43,41 @@ class Ref
      */
     unsigned int getRefCount() const;
 
+    /**
```

Comment:
This isn't actually a smart pointer -- you can't use it as a pointer to a Ref. The sole purpose is to hold a reference to an object while calling into "other" code that may release a reference.
The reason for this is that code that has multiple exit paths is a bug waiting to happen if you don't have exception-safe stack-based clean-up. I've seen that happen in so many code bases, it's not even funny, and the stack-based reference count (different from smart pointer) makes that class of bug go away.